packages/devtools/src/sourceMap/loadOriginalSourceLoc.ts

```diff
@@ -0,0 +1,178 @@
+import { getFilePath, getOriginalPosition, resources } from './sourceMapConsumer';
+
+// when only one source map appears in a source file: source url -> source map JSON
+// when more than one source maps appear in a source file: `${source}:${line}:${column}` -> source map JSON
+const sourceMapJSONs: Map<string, RawSourceMap> = new Map();
+
+const getComputedSouceMapJSON = ({ source, line, column }: MappedPosition) => {
+  return sourceMapJSONs.get(source) ?? sourceMapJSONs.get(`${source}:${line}:${column}`);
+};
+
+export async function loadOriginalSourceLoc(sourceUrlWithLoc: string): Promise<MappedPosition> {
+  const paths = sourceUrlWithLoc.split(':');
+  const column = Number(paths.pop());
+  const line = Number(paths.pop());
+  const source = paths.join(':');
+  if (Number.isNaN(line) || Number.isNaN(column)) {
+    return { source, line: 1, column: 0 };
+  }
+  const sourceLoc: MappedPosition = { source, line, column };
+
+  const sourceMapJSON = getComputedSouceMapJSON(sourceLoc);
+  if (sourceMapJSON) {
+    return getOriginalPosition(sourceMapJSON, sourceLoc);
+  }
+
+  try {
+    return extractAndLoadSourceMapJSON(sourceLoc);
+  } catch (error) {
+    console.warn(`[Griffel devtools] unable to load source map for ${sourceUrlWithLoc}`);
+    console.warn(error);
+    return sourceLoc;
+  }
+}
+
+// inspired by https://github.com/facebook/react/blob/b66936ece7d9ad41a33e077933c9af0bda8bff87/packages/react-devtools-shared/src/hooks/parseHookNames/loadSourceAndMetadata.js#L135
+async function extractAndLoadSourceMapJSON(sourceLoc: MappedPosition): Promise<MappedPosition> {
+  const { source, line, column } = sourceLoc;
+
+  const runtimeSourceCode = await fetchFiles(source);
+
+  const externalSourceMapURLs = [];
+  let sourceMapNum = 0;
+  const sourceMapRegex = / ?sourceMappingURL=([^\s'"]+)/gm;
+
+  let sourceMappingURLMatch = sourceMapRegex.exec(runtimeSourceCode);
+  if (!sourceMappingURLMatch) {
+    throw new Error(`[Griffel devtools] extractAndLoadSourceMapJSON() no source map found in ${source}`);
+  }
+
+  while (sourceMappingURLMatch) {
+    sourceMapNum++;
+    const sourceMappingURL = sourceMappingURLMatch[1];
+
+    const hasInlineSourceMap = sourceMappingURL.indexOf('base64,') >= 0;
+    if (hasInlineSourceMap) {
+      try {
+        // Web apps like Code Sandbox embed multiple inline source maps.
+        // In this case, we need to loop through and find the right one.
+        const trimmed = sourceMappingURL.match(/base64,([a-zA-Z0-9+/=]+)/)?.[1] ?? '';
+        const decoded = decodeBase64String(trimmed);
+        const sourceMapJSON = JSON.parse(decoded);
+
+        // TODO optionally turn on debug message
+        // if (__DEBUG__) {
+        //   console.groupCollapsed('[Griffel devtools] extractAndLoadSourceMapJSON() Inline source map');
+        //   console.log(sourceMapJSON);
+        //   console.groupEnd();
+        // }
+
+        if (sourceMapIncludesSource(sourceMapJSON, getFilePath(source))) {
```

Comment:
This code to insert into the map seems super weird with the usage of `sourceMapNum`, couldn't you just collect all entries in the loop and insert at the end ? it would simplify this code a lot